psh/tools/common.py

```diff
@@ -20,6 +23,14 @@
 # acceptable separators: white spaces (wss) + CC, CC + wss, wss
 SEPARATOR_PATTERN = r'(?:' + CONTROL_CODE + r'|\s)+'
 
+# When subtracting sets the order is mixed, we want to avoid this
+CHARS = sorted(list(set(string.printable) - set(string.whitespace) - set('/')))
```

Comment:
Why do we care about the order?
I think `CHARS` might be too general for this set of chars (I would expect that chars are from range 0-255). Maybe we do not need to define it here, just put it directly in the tests

Reply:
I am not sure whether randomness is a good thing here. There were random cases (1~50), which revealed the https://github.com/phoenix-rtos/phoenix-rtos-project/issues/521 issue, but I think it's better to have separate test case for checking that.

About name: I don't have an idea for some better common name here.

It will be used in more tests, so I think it's better to place it here.

Reply:
I still don't understand why we need to sort the `CHARS` variable. As I look at the tests, we use `CHARS` in two cases:
1. In the line 36 as an argument to the `history` call
2. As an argument for `get_rand_strings`

In the second case, the order of the pool doesn't matter. Since we are making the random selection of elements from the pool we don't care about the order (we treat the pool as a box with a hole from which we pull out a ball with some letter written on it and then put the ball back in the box).
In the first case if the order is important we should just pass `sorted(CHARS)` there and leave a comment why we are doing this.

If random testing reveals any problems then we're happy and leave it with the hope that this test case might help us find some other weird bugs in the future. (But maybe I did not understand your reply correctly. If there is something I don't see, please correct me)

About the variable name: OK, you can leave it as is. I also do not have the good name for this variable. I made the name `CHARS` in the `mkdir` test because it was defined locally in the module.